index.js

```diff
@@ -5,20 +5,36 @@
 
 'use strict'
 
-var EventEmitter = require('events').EventEmitter
-var util = require('util')
-var preBuild = require('./lib/pre-build')
-var natives = preBuild.load('native_metrics')
-
-var DEFAULT_TIMEOUT = 15 * 1000 // 15 seconds
-var GC_TYPE_NAMES = {
-  1: 'Scavenge',
-  2: 'MarkSweepCompact',
-  4: 'IncrementalMarking',
-  8: 'ProcessWeakCallbacks',
-
-  3: 'All', // Node v4 and earlier only have Scavenge and MarkSweepCompact.
-  15: 'All'
+const EventEmitter = require('events').EventEmitter
+const util = require('util')
+const preBuild = require('./lib/pre-build')
+const natives = preBuild.load('native_metrics')
+const semver = require('semver')
+
+const DEFAULT_TIMEOUT = 15 * 1000 // 15 seconds
+let GC_TYPE_NAMES = null
+
+// In Node 18(v8 10) the GCType enum added `MinorMarkCompact`
+// we have to update our mapping to properly account for this
+if (semver.satisfies(process.version, '>=18')) {
```

Comment:
this is the fix. apparently i upgraded eslint config and it converted all `var` to `const`/`let`.  I can revert and open separate PR for that if need be